packages/utils/test/formatingSpace.test.ts

```diff
@@ -0,0 +1,303 @@
+            const nodes = htmlToElements(`<p><a>abc
+     </a></p>`);
+            expect(removeFormattingSpace(nodes[0].childNodes[0].childNodes[0])).to.equal('abc');
+        });
+        it('(08) Should return trimmed text when spaces and line breaks before text', async () => {
+            const nodes = htmlToElements(`<p><a>
+     abc</a></p>`);
+            expect(removeFormattingSpace(nodes[0].childNodes[0].childNodes[0])).to.equal('abc');
+        });
+        it('(09) Should return trimmed text when multiples spaces before and after node', async () => {
+            const nodes = htmlToElements('<p>    <a>abc</a>    </p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(10) Should return trimmed text when multiple spaces after node', async () => {
+            const nodes = htmlToElements('<p><a>abc</a>   </p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[0].childNodes[0])).to.equal('abc');
+        });
+        it('(11) Should return trimmed text when multiple spaces before node', async () => {
+            const nodes = htmlToElements('<p>   <a>abc</a></p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(12) Should return trimmed text when multiples spaces before and after node and text', async () => {
+            const nodes = htmlToElements('<p>    <a>  abc  </a>    </p>');
+            expect(removeFormattingSpace(nodes[0].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(13) Should return trimmed text when multiples spaces and line breaks before and after node and text', async () => {
+            const nodes = htmlToElements(`<p> 
+    <a>  abc  </a>  
+</p>`);
+            expect(removeFormattingSpace(nodes[0].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(14) Should return trimmed text when multiple spaces after both nodes (alt)', async () => {
+            const nodes = htmlToElements('<p><a>abc</a>   </p>   ');
+            expect(removeFormattingSpace(nodes[0].childNodes[0].childNodes[0])).to.equal('abc');
+        });
+        it('(15) Should return trimmed text when multiple spaces before both node (alt)', async () => {
+            const nodes = htmlToElements('   <p>   <a>abc</a></p>');
+            expect(removeFormattingSpace(nodes[1].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(16) Should return trimmed text when multiples spaces and line breaks before and after node and text (alt)', async () => {
+            const nodes = htmlToElements(`
+  <p> 
+    <a>  abc  </a>  
+  </p>
+`);
+            expect(removeFormattingSpace(nodes[1].childNodes[1].childNodes[0])).to.equal('abc');
+        });
+        it('(17) should preserve a space before a line break at the end of a paragraph', async () => {
```

Comment:
Now we have one test with a line break at the end. Maybe you could add some, with space before and after the line break (you know how weird it is that the space before the line break is treated differently from the space after...).

Reply:
I can do that